### DISCUSSION TITLE
Create 3 application with BLE beacons and IR LEDs.

### DISCUSSION THREAD

**Discussion by gmptrr98** [iRobotEducation/create3_docs#298] [Show and tell]
Firstly I would like to thank you all for the help given trough this Github page. 

I completed my thesis with the following project, using Create 3 and a RaspberryPi to run the developed custom nodes. BLE beacons are used to start and stop the operation as well as computing a rough estimate of the position of the goal. The goal, modeled by a small wooden table, is equipped with an Arduino Mega and 2 IR LEDs. If you have any question I'll be happy to answer, and if you are particularly interested I can share with you my thesis!

The video is just 1:38 minutes long and shows a test of this algorithm.
 https://youtu.be/5ZpJ1xOYV0I

I would also like to thank you for giving us a product like Create 3, these few months working with it have been both fun and very rewarding, as it gave me the opportunity to learn practical skills I've always wanted to develop.

**Comment by shamlian** [comment]
Congratulations on completing your thesis! I think we'd love to read it (I'll speak for myself -- I would love to); if you'd rather not post it publicly, you could send it to education@irobot.com and it will find its way to the team. Could you explain what the clicker device is, and how it relates to the location of the goal object? Is the robot using BLE for coarse localization and then IR to servo in? We love seeing the cool stuff you have built, thanks for posting!

**Reply by gmptrr98** [reply]
I'll send my thesis on the email, if you have trouble finding it I can send it to you directly. 
The clicker device is a Blue Charm beacon, when it is double pressed it is turned on for 300 seconds. The RaspberryPi runs a node checking for this particular beacon, if it is on the robot starts looking for the goal. Usually this beacon would be positioned directly on the goal (the "wooden table", which actually is a mockup for a Powered Wheelchair). There 2 more beacons placed in known positions in the environment. Using RSSI it is possible to estimate the position of the "wheelchair", considering that the robot starts from the dock, which is position x=0, y=0. Theoretically, with newer beacons coming up with Bluetooth 5.1 just a single beacon would be enough to tell distance and direction, but right now I had to use trilateration techniques which weren't as precise as I wished. 
When the IR are sensed the robot starts approaching the model (in the video you can see the light ring switch from orange to blue). It functions similarly to the Roomba's dock. The main difference is that, since the LEDs are positioned at an height of 17 cm, the robot loses track of them when very close. This is intended: once this happens the robot goes to the "Entering" state (purple ring), rotates to the last orientation a 172 opcode was detected, and goes forward. 
I hope my explanation is a clear enough, if you have any other question I would be happy to answer! If you have the chance to read the thesis let me know what you think, I believe you would find the final intended application very interesting!

**Comment by vivianweli** [comment]
Hi gmptrr98, we're working on an undergraduate project similar to yours also utilizing BLE beacons and a Pi. We are really interested in reading your thesis to help guide us in our process. I've sent you an invitation to connect on LinkedIn if that works for you. Thanks!
